pkg/sriov/sriov.go

```diff
@@ -18,10 +19,17 @@ import (
 type NetlinkManager interface {
 	LinkByName(string) (netlink.Link, error)
 	LinkSetVfVlan(netlink.Link, int, int) error
+	LinkSetVfHardwareAddr(netlink.Link, int, net.HardwareAddr) error
+	LinkSetHardwareAddr(netlink.Link, net.HardwareAddr) error
 	LinkSetUp(netlink.Link) error
 	LinkSetDown(netlink.Link) error
 	LinkSetNsFd(netlink.Link, int) error
 	LinkSetName(netlink.Link, string) error
+	LinkSetVfTxRate(netlink.Link, int, int) error
+	LinkSetVfSpoofchk(netlink.Link, int, bool) error
+	LinkSetVfTrust(netlink.Link, int, bool) error
+	LinkSetVfNodeGUID(netlink.Link, int, net.HardwareAddr) error
+	LinkSetVfPortGUID(netlink.Link, int, net.HardwareAddr) error
```

Comment:
Do we need the LinkSetVfTxRate, LinkSetVfNodeGUID, LinkSetVfPortGUID? What is the point of this PRto expose all possible setting all the vf? What is the use-case for this PR?

Reply:
Hi, 
Good questions, thank you!

The use case for this came about when testing DPDK/SRIOV with the testpmd app. By default, testpmd uses hardcoded MAC addresses. As such, the testpmd app was not able to pass packets, as the hardcoded MAC addresses didn't match the VF MAC addresses. So, I had to disable spoofchk for the packets to pass. 

While I was adding support for the spoofchk flag, I thought it would be good to add additional flags that are currently supported in the netlink lib. So, yes, the point of this PR would be to expose all possible, currently supported (netlink and device) VF flags. In the current state, this PR only adds support for a subset of the available VF flags (from "man ip link").

Now, DPDK based apps, including testpmd, have the command options, such as whitelist, to specify the MAC(s) to use, which would not require disabling spoofchk. 

But, the thinking is that there may be use cases where a DPDK app may desire to use a MAC other than the assigned MAC, which would require spoofchk to be disabled. 

Hope this is helpful, thanks for the input and comments.
Vic

Reply:
I understand the use-case for spoofcheck and trust mode, but I still wonder if we need to expose all possible VF flags or just add them when we have real use-case.

Reply:
Sure, that is an option also. I can adjust this PR to only set those flags. I'm not aware of real-world use cases for the other flags, so I'm OK with that approach as well.

Any thoughts from others regarding just supporting spoofcheck and trust mode VF flags?

Reply:
Although, I think the max_tx_rate may be useful, for rate limiting

Reply:
Yes I agree regarding the max_tx_rate we also added PR to https://github.com/vishvananda/netlink/pull/469 to add min rate so once it merged you can add that too.

Reply:
Ok, sounds good. I will add the min_rate once that PR is merged